src/EventListener/AssetsInstallCommandListener.php

```diff
@@ -245,15 +239,10 @@ private function hardCopy(string $originDir, string $targetDir): string
         return self::METHOD_COPY;
     }
 
-    private function getPublicDirectory(ContainerInterface $container): string
+    private function getPublicDirectory(): string
     {
         $defaultPublicDir = 'public';
-
-        if (null === $this->projectDir && !$container->hasParameter('kernel.project_dir')) {
-            return $defaultPublicDir;
-        }
-
-        $composerFilePath = ($this->projectDir ?? $container->getParameter('kernel.project_dir')).'/composer.json';
```

Comment:
why remove all of this and what has to do with psalm?

Reply:
projectDir cannot be null now